### PR TITLE
Use native SSL stack in scraper for Windows builds

### PR DESCRIPTION
**Pull request by cyrossignol**
From Slack:
> IIRC on Windows we can build libcurl `-with-winssl` to use the native TLS stack (instead of openssl) that verifies certs against the native CA bundle in the Windows certificate store. Windows update keeps these current...not sure if it's a good idea to compile the bundle into the program--if we have scrapers running on both Windows an Linux, we may have trouble reaching convergence if the upstream cert for one of the projects gets revoked or renewed by a recent authority not in the bundle compiled into the wallet

OpenSSL can hook into some features of WinSSL but uses its own CA certificate bundle. Note the use of OpenSSL in the current build of curl:

```
$ ./curl.exe --version
curl 7.63.0 (x86_64-w64-mingw32) libcurl/7.63.0 OpenSSL/1.0.1k (WinSSL)
...
Features: AsynchDNS IPv6 Largefile SSPI Kerberos SPNEGO NTLM SSL TLS-SRP HTTPS-proxy MultiSSL
```

By disabling OpenSSL when configuring it, curl defaults to the native Windows stack so we don't need to distribute a certificate bundle:

```
$ ./curl.exe --version
curl 7.63.0 (x86_64-w64-mingw32) libcurl/7.63.0 WinSSL
...
Features: AsynchDNS IPv6 Largefile SSPI Kerberos SPNEGO NTLM SSL
```

### CODE REVIEW

File: depends/packages/openssl.mk [hunk 1]
```diff
@@ -30,7 +30,7 @@ $(package)_config_opts+=no-rfc3779
 $(package)_config_opts+=no-rsax
 $(package)_config_opts+=no-sctp
 $(package)_config_opts+=no-seed
-# $(package)_config_opts+=no-sha0
+$(package)_config_opts+=no-sha0
```
Comment (cyrossignol) [inline depends/packages/openssl.mk]: Note: this goes away when we get to OpenSSL 1.1.x

**Comment by cyrossignol** [comment]
We can also use macOS's native Secure Transport API by declaring `--with-darwinssl`. Not sure if this is strictly necessary.

**Comment by jamescowens** [comment]
Probably should as the analogue to Windows...

**Comment by cyrossignol** [comment]
Seems to be working, including WCG: 
```
02/25/19 03:37:16 INFO: Scraper: <DownloadProjectTeamFiles>: Downloading project file for World_Community_Grid
02/25/19 03:37:17 INFO: Scraper: <DownloadProjectTeamFiles>: Successfully pulled team header file for World_Community_Grid
02/25/19 03:37:18 INFO: Scraper: <ProcessProjectTeamFile>: Opening team file (C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-team.gz)
02/25/19 03:37:18 INFO: Scraper: <ProcessProjectTeamFile>: Started processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-team.gz
02/25/19 03:37:19 INFO: Scraper: <ProcessProjectTeamFile>: Persisting Team ID entries to disk.
02/25/19 03:37:19 INFO: Scraper: <StoreTeamIDList>: Started processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\TeamIDs.csv.gz
02/25/19 03:37:19 INFO: Scraper: <StoreTeamIDList>: Finished processing Team ID data from map.
02/25/19 03:37:19 INFO: Scraper: <StoreTeamIDList>: Process Complete.
02/25/19 03:37:19 INFO: Scraper: <ProcessProjectTeamFile>: Stored Team ID entries.
02/25/19 03:37:19 INFO: Scraper: <ProcessProjectTeamFile>: Finished processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-team.gz
```
```
02/25/19 03:41:48 INFO: Scraper: <DownloadProjectRacFiles>: Downloading project file for World_Community_Grid
02/25/19 03:41:49 INFO: Scraper: <DownloadProjectRacFiles>: Successfully pulled rac header file for World_Community_Grid
02/25/19 03:42:03 INFO: Scraper: <ProcessProjectRacFileByCPID>: Opening rac file (C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-user.gz)
02/25/19 03:42:03 INFO: Scraper: <ProcessProjectRacFileByCPID>: Started processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-user.gz
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Finished processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-user.gz
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: FileHash by GetFileHash 065734ab9329249e12952dccbf991dbd72d58f544634468d7bc68086d50c1f2d
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Processing new rac file C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-user.gz(36666581 -> 1031)
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Created manifest entry for 065734ab9329249e12952dccbf991dbd72d58f544634468d7bc68086d50c1f2d C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\World_Community_Grid-2f7cd5-582acd95dea85.csv.gz
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Persisting manifest entry to disk.
02/25/19 03:42:06 INFO: Scraper: <StoreScraperFileManifest>: Started processing C:\Users\Cy\dotfiles\temp\gridcoin\testnet\Scraper\Manifest.csv.gz
02/25/19 03:42:06 INFO: Scraper: <StoreScraperFileManifest>: Finished processing manifest from map.
02/25/19 03:42:06 INFO: Scraper: <StoreScraperFileManifest>: Process Complete.
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Stored Manifest
02/25/19 03:42:06 INFO: Scraper: <ProcessProjectRacFileByCPID>: Complete Process
```

File: depends/packages/openssl.mk (hunk 1 above)
Comment (jamescowens) [reply depends/packages/openssl.mk]: Yeah. And we shouldn't be using anything based on SHA0 or 1...

**Comment by jamescowens** [comment]
Looks good to me. I added Ravon to the reviewers to meet the two person review rule. I think you can change your PR from draft to mergeable!

**Comment by denravonska** [comment]
> Seems to be working, including WCG:
> ...

Was this on Windows or OSX?

**Comment by cyrossignol** [comment]
@denravonska That was on Windows... haven't tried mac.

**Comment by jamescowens** [comment]
Merging. Ravon will try on mac after merge.

**Pull request merged**